AnkiDroid/src/main/java/com/ichi2/anki/Whiteboard.java

```diff
@@ -120,17 +128,38 @@ public Whiteboard(AbstractFlashcardViewer cardViewer, boolean inverted) {
         mBitmapPaint = new Paint(Paint.DITHER_FLAG);
 
         // selecting pen color to draw
-        mColorPalette = cardViewer.findViewById(R.id.whiteboard_editor);
+        mColorPalette = activity.findViewById(R.id.whiteboard_editor);
 
-        cardViewer.findViewById(R.id.pen_color_red).setOnClickListener(this::onClick);
-        cardViewer.findViewById(R.id.pen_color_green).setOnClickListener(this::onClick);
-        cardViewer.findViewById(R.id.pen_color_blue).setOnClickListener(this::onClick);
-        cardViewer.findViewById(R.id.pen_color_yellow).setOnClickListener(this::onClick);
-        cardViewer.findViewById(R.id.stroke_width).setOnClickListener(this::onClick);
+        activity.findViewById(R.id.pen_color_red).setOnClickListener(this::onClick);
+        activity.findViewById(R.id.pen_color_green).setOnClickListener(this::onClick);
+        activity.findViewById(R.id.pen_color_blue).setOnClickListener(this::onClick);
+        activity.findViewById(R.id.pen_color_yellow).setOnClickListener(this::onClick);
+        activity.findViewById(R.id.stroke_width).setOnClickListener(this::onClick);
     }
 
     public int getCurrentStrokeWidth() {
-        return AnkiDroidApp.getSharedPrefs(mCardViewer.get()).getInt("whiteBoardStrokeWidth", 6);
+        return AnkiDroidApp.getSharedPrefs(mAnkiActivity).getInt("whiteBoardStrokeWidth", 6);
+    }
+
+
+    public static Whiteboard createInstance(AnkiActivity context, HandleTouch handleTouchControl, WhiteboardMethods abstractFlashcardViewerUI) {
+
+        SharedPreferences sharedPrefs = AnkiDroidApp.getSharedPrefs(context);
+        Whiteboard whiteboard = new Whiteboard(context, handleTouchControl, isInNightMode(sharedPrefs));
+
+        if (abstractFlashcardViewerUI != null) {
+            mAbstractFlashcardViewerUI = abstractFlashcardViewerUI;
+        }
```

Comment:
Also, change the name, and add documentation 
```suggestion
        mAbstractFlashcardViewerUI = abstractFlashcardViewerUI
```